### ISSUE TITLE
Allow user to change wallet password

### ISSUE THREAD

**Issue by onyb** [brave/brave-browser#29287]
|Community|https://community.brave.com/t/password-reset-available/478442|
-|-

We need a new method in `KeyringService` to change the wallet password. The current way of restoring by creating a new wallet with a different password is not ideal.

cc: @darkdh
cc: @kdenhartog FYI

**Comment by darkdh** [comment]
Agreed, we should expose a mojo API that decrypt all encrypted with old password and re-encrypt them with new password and write to the pref.

**Comment by kdenhartog** [comment]
+1 agreed this is a good idea.

**Comment by christiantjmdcity** [comment]
+1 agreed this is a good idea.

**Comment by ParamaShiva1** [comment]
It will be great. Yes. Thanks!

**Comment by DevVali** [comment]
Any updates on this? Would love to see this feature!

**Comment by girlwithnoname77** [comment]
This would have been useful as I created a new account and the password didn't save properly in my password manager before moving on to create the account. Very frustrating.
